src/main/java/bio/terra/service/dataset/flight/ingest/IngestValidateGcpRefsStep.java

```diff
@@ -35,13 +36,16 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
     // Then probe the file system to validate that the file exists and is part
     // of this dataset. We check all ids and return one complete error.
 
-    Set<String> invalidRefIds = new HashSet<>();
+    Set<InvalidRefId> invalidRefIds = new HashSet<>();
     for (Column column : table.getColumns()) {
       if (column.isFileOrDirRef()) {
         List<String> refIdArray = bigQueryPdao.getRefIds(dataset, stagingTableName, column);
         List<String> badRefIds = fileDao.validateRefIds(dataset, refIdArray);
         if (badRefIds != null) {
-          invalidRefIds.addAll(badRefIds);
+          invalidRefIds.addAll(
+              badRefIds.stream()
+                  .map(id -> new InvalidRefId(id, column.getName()))
+                  .collect(Collectors.toList()));
```

Comment:
This builds a collection and then adds it to a set. It should be more efficient to add each element to the set directly instead of first building a collection.
```suggestion
          badRefIds.forEach(id -> badRefIds.add(new InvalidRefId(id, column.getName()));
```